### PR TITLE
Implement wpa_supplicant test

### PR DESCRIPTION
**Pull request by grisu48**
Adds a test for wpa_supplicant. The wpa_supplicant test is a
self-sufficient zip archive that gets extracted and executed.

The wpa_supplicant test creates two virtual wlan interfaces and tests
the wlan functionality by setting up a hotspot via hostapd.
wpa_supplicant is then tested, if able to obtain a network list and
able to connect to an open and encrypted network. Lastly we test if
wicked is able to get a ip address via dhcp on the wlan interface.

- Related ticket: https://progress.opensuse.org/issues/55715
- Needles: N/A
- Verification run: [SLE 15-SP1](https://openqa.suse.de/tests/4096488) | [SLE 15](https://openqa.suse.de/tests/4096489) | [Tumbleweed](http://phoenix-openqa.qam.suse.de/tests/391) | [Leap 15.2](http://phoenix-openqa.qam.suse.de/tests/392)

This test is only available on `x86_64` and SLE 15+.

### CODE REVIEW

**Comment by okurz** [comment]
Doesn't this duplicate the functionality of the already available test using hostapd and wpa_supplicant in https://github.com/os-autoinst/os-autoinst-distri-opensuse/tree/master/tests/x11/network ?

**Comment by grisu48** [comment]
> Doesn't this duplicate the functionality of the already available test using hostapd and wpa_supplicant in https://github.com/os-autoinst/os-autoinst-distri-opensuse/tree/master/tests/x11/network ?

Hi Oliver! I knew this was coming after the email this morning ;-)

In my opinion it doesn't: https://github.com/os-autoinst/os-autoinst-distri-opensuse/tree/master/tests/x11/network tests for the integration of hostapd and NetworkManager, while this tests the functionality of `wpa_supplicant`. Both use `hostapd` but test a different wpa client.

File: lib/main_common.pm [hunk 1]
```diff
@@ -1710,6 +1710,7 @@ sub load_extra_tests_console {
     loadtest 'console/libgcrypt' if ((is_sle(">=12-SP4") && (check_var_array('ADDONS', 'sdk') || check_var_array('SCC_ADDONS', 'sdk'))) || is_opensuse);
     loadtest "console/gd";
     loadtest 'console/valgrind' unless is_sle('<=12-SP3');
+    loadtest 'console/wpa_supplicant' if (is_x86_64 && (is_sle('15+') || is_leap('15.1+') || is_tumbleweed));
```
Comment (okurz) [inline lib/main_common.pm:1713]: Our tests should always consider openSUSE Tumbleweed as the default product without a version. That means any differing behaviour should have an explicit exclude rule for the "older products" and potentially an explanation why the behaviour should differ. You can try to negate the logic check.

Comment (grisu48) [reply lib/main_common.pm:1713]: I'll change this issue in the afternoon, thanks for reminding me about the logic.

I understand your point about the test suites but don't have a sufficient overview yet. Do you have a recommendation where to put the test? Would be happy to apply the changes then.

Comment (grisu48) [reply lib/main_common.pm]: Changed it to `loadtest 'console/wpa_supplicant' unless (!is_x86_64 || is_sle('<15') || is_leap('<15.1'));`

**Comment by grisu48** [comment]
Rebased it so it can be merged.

**Comment by okurz** [comment]
+1

**Comment by DimStar77** [comment]
https://openqa.opensuse.org/tests/1236322

Fails on JeOS

**Comment by DimStar77** [comment]
> https://progress.opensuse.org/issues/55715

And: can we please stop with non public progress tickets??

**Comment by Vogtinator** [comment]
> https://openqa.opensuse.org/tests/1236322
> 
> Fails on JeOS

Most likely because of `modprobe mac80211_hwsim` not being available with kernel-default-base, so just disabling this test would be the right fix. I'll make a PR.

**Pull request merged**